ext/geos_c_impl/factory.c

```diff
@@ -44,7 +44,7 @@ notice_handler(const char* fmt, ...)
 #endif
 
 static void
-error_handler(const char* fmt, ...)
+NORETURN(error_handler)(const char* fmt, ...)
```

Comment:
What's the purpose of this? Never seen it before so just curious.

Reply:
It is a ruby implementation version of the C23 `noreturn` keyword (C17 ~ `_Noreturn` in some compilers) that tells the compilers that this function will never return.

This helps get rid of a warning (`warning: function 'error_handler' could be declared with attribute 'noreturn' [-Wmissing-noreturn]`) and very probably helps the compiler optimizing.